src/External/Presentation/GoalFinder.WebApi/Endpoints/User/ReportUserAfterMatch/ReportUserAfterMatchEndpoint.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using FastEndpoints;
+using GoalFinder.Application.Features.User.GetUserInfoOnSidebar;
+using GoalFinder.Application.Features.User.ReportUserAfterMatch;
+using GoalFinder.Application.Features.User.UpdateUserInfo;
+using GoalFinder.WebApi.Endpoints.User.ReportUserAfterMatch.Common;
+using GoalFinder.WebApi.Endpoints.User.ReportUserAfterMatch.HttpResponseMapper;
+using GoalFinder.WebApi.Endpoints.User.ReportUserAfterMatch.Middleware.Authorization;
+using GoalFinder.WebApi.Endpoints.User.ReportUserAfterMatch.Middleware.Caching;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Http;
+
+namespace GoalFinder.WebApi.Endpoints.User.ReportUserAfterMatch;
+
+/// <summary>
+///     Endpoint for report user after match
+/// </summary>
+internal sealed class ReportUserAfterMatchEndpoint
+    : Endpoint<ReportUserAfterMatchRequest, ReportUserAfterMatchHttpResponse>
+{
+    public override void Configure()
+    {
+        Patch(routePatterns: "user/report");
+        AuthSchemes(authSchemeNames: JwtBearerDefaults.AuthenticationScheme);
+        DontThrowIfValidationFails();
+        PreProcessor<ReportUserAfterMatchAuthorizationPreProcessor>();
+        PreProcessor<ReportUserAfterMatchCachingPreProcessor>();
+        PostProcessor<ReportUserAfterMatchCachingPostProcessor>();
+        Description(builder: builder =>
+        {
+            builder.ClearDefaultProduces(
+                StatusCodes.Status400BadRequest,
+                StatusCodes.Status401Unauthorized,
+                StatusCodes.Status403Forbidden
+            );
+        });
+        Summary(endpointSummary: summary =>
+        {
+            summary.Summary = "Endpoint for update prestige score after each match.";
+            summary.Description =
+                "This endpoint is used for update prestige score after each match.";
+            summary.ExampleRequest = new() { 
+                FootballMatchId = Guid.Empty,
+                UserId = Guid.Empty,
+                PlayerScores = new List<PlayerPrestigeScore>
+                {
+                    new PlayerPrestigeScore { PlayerId = Guid.Empty, BonusScore = 20 },
+                    new PlayerPrestigeScore { PlayerId = Guid.Empty, BonusScore = -20 }
```

Comment:
anh góp ý trước logic **hiện tại của em nha**
chỗ này anh thấy logic oke hơn thì nên chỉ nhận data kiểu report xấu ( bị trừ điểm ) 
bên FE auto em vào thì giao diện "xanh hết" => em bấm oke luôn thì nó gửi form null về => nghĩa là không ai bị report -> cộng điểm hết cho mấy người khác 
thì thay vì em mỗi lần gửi 1 đống data thì chỉ cần gửi mấy cái bị report xấu thôi

Còn bàn đến chuyện yêu cầu anh Thắng trong cái docx requirement thì ảnh bảo xét trên tiêu chí "nhiều người report xấu" mới trừ ( kiểu giống như bỏ phiếu hơn là report trực tiếp ) => mà anh xem lại thì có vẻ team BE đã hiểu sai chỗ ni nên db nó đang sai sót. 
*Cách xử lý cũng khá dễ 
c1 : là ở bảng người tham gia trong trận bóng thì cái trường điểm cộng đổi thành string đổi tên nó thành "badReportList" các kiểu,... 
>> Khi 1 user report thì thêm nó vào string -> ví dụ : "trinh,hoang,.." mấy cái này là userName
xong thì tổng kết bằng cách cắt chuỗi bằng , rồi dùng string length mà trừ điểm
c2: giữ nguyên kiểu dữ liệu => 1 người report xấu thì cộng thêm 1 

=> thì chung quy kết quả của 2 cách trên em có 1 cái mốc để biết được nó đã đến lúc bị trừ điểm chưa